### ISSUE TITLE
Anonymous classes in mixins no longer work

### ISSUE THREAD

**Issue by Runemoro** [FabricMC/Mixin#12]
They used to work properly on Forge, but they now throw `NoClassDefFoundError`s.

**Comment by asiekierka** [comment]
Log?

**Comment by Runemoro** [comment]
```
Exception in thread "Client thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:132)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:130)
	... 1 more
Caused by: java.lang.NoClassDefFoundError: net/minecraft/block/Block$Anonymous$5364f9631be94562b731d7dde2f2818e
	at net.minecraft.block.Blocks.<clinit>(Blocks.java:698)
	at net.minecraft.util.registry.Registry.method_10237(Registry.java:83)
	at net.minecraft.util.registry.Registry.method_10238(Registry.java:234)
	at java.util.LinkedHashMap$LinkedEntrySet.forEach(LinkedHashMap.java:671)
	at net.minecraft.util.registry.Registry.<clinit>(Registry.java:233)
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:34)
	at net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:352)
	at net.minecraft.client.main.Main.main(Main.java:127)
	... 6 more
Caused by: java.lang.ClassNotFoundException: net.minecraft.block.Block$Anonymous$5364f9631be94562b731d7dde2f2818e
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:148)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 14 more
```

Here's the mixin: https://gist.github.com/Runemoro/02bf2f7e273b415dd0bdb4aaec2a75c4

**Comment by asiekierka** [comment]
Thanks; I needed to know what string to check for (to avoid the performance penalty of going through Mixin every time).

**Comment by asiekierka** [comment]
Loader-side issue, *should* be fixed in 0.4.5.
